wal/wal.go

```diff
@@ -172,11 +172,8 @@ func openAtIndex(dirpath string, snap walpb.Snapshot, all bool) (*WAL, error) {
 		}
 		err = l.TryLock()
 		if err != nil {
-			if all {
+			if write {
 				return nil, err
-			} else {
```

Comment:
the error from l.TryLock().
It could be either fileutil.ErrLocked or the error from syscall.